### ISSUE TITLE
L515 : An error has occurred during frame callback: map::at

### ISSUE THREAD

**Issue by vinesmsuic** [realsenseai/realsense-ros#1872]
# Problem

An Error log message generated in every frame and flood the shell output.
The `align_depth` topics are available and the image can be seen on `rqt_gui`.
Although this problem  did not cause much noticeable error on performance, I found it disturbing and it generates a huge size of log file.
This error only occur when `align_depth` is enabled.

# Environment
Device: Nvidia Jetson Xavier NX, RealSense LiDAR L515
OS: Ubuntu 18.04
ROS Version: ROS Melodic
RealSense SDK Version: LibRealSense SDK v2.41.0
realsense-ros Version: 2.2.21

# To reproduce the Error

My `rs_camera.launch` file
```xml
<launch>
  <arg name="serial_no"           default=""/>
  <arg name="usb_port_id"         default=""/>
  <arg name="device_type"         default=""/>
  <arg name="json_file_path"      default=""/>
  <arg name="camera"              default="camera"/>
  <arg name="tf_prefix"           default="$(arg camera)"/>
  <arg name="external_manager"    default="false"/>
  <arg name="manager"             default="realsense2_camera_manager"/>
  <arg name="output"              default="screen"/>

  <arg name="fisheye_width"       default="640"/>
  <arg name="fisheye_height"      default="480"/>
  <arg name="enable_fisheye"      default="false"/>

  <arg name="depth_width"         default="640"/>
  <arg name="depth_height"        default="480"/>
  <arg name="enable_depth"        default="true"/>

  <arg name="confidence_width"    default="640"/>
  <arg name="confidence_height"   default="480"/>
  <arg name="enable_confidence"   default="true"/>
  <arg name="confidence_fps"      default="30"/>

  <arg name="infra_width"         default="640"/>
  <arg name="infra_height"        default="480"/>
  <arg name="enable_infra"        default="true"/>
  <arg name="enable_infra1"       default="false"/>
  <arg name="enable_infra2"       default="false"/>
  <arg name="infra_rgb"           default="false"/>

  <arg name="color_width"         default="1280"/>
  <arg name="color_height"        default="720"/>
  <arg name="enable_color"        default="true"/>

  <arg name="fisheye_fps"         default="30"/>
  <arg name="depth_fps"           default="30"/>
  <arg name="infra_fps"           default="30"/>
  <arg name="color_fps"           default="30"/>
  <arg name="gyro_fps"            default="400"/>
  <arg name="accel_fps"           default="250"/>
  <arg name="enable_gyro"         default="true"/>
  <arg name="enable_accel"        default="true"/>

  <arg name="enable_pointcloud"         default="false"/>
  <arg name="pointcloud_texture_stream" default="RS2_STREAM_COLOR"/>
  <arg name="pointcloud_texture_index"  default="0"/>

  <arg name="enable_sync"               default="true"/>
  <arg name="align_depth"               default="true"/>

  <arg name="publish_tf"                default="true"/>
  <arg name="tf_publish_rate"           default="0"/>

  <arg name="filters"                   default=""/>
  <arg name="clip_distance"             default="-2"/>
  <arg name="linear_accel_cov"          default="0.01"/>
  <arg name="initial_reset"             default="false"/>
  <arg name="unite_imu_method"          default="copy"/>
  <arg name="topic_odom_in"             default="odom_in"/>
  <arg name="calib_odom_file"           default=""/>
  <arg name="publish_odom_tf"           default="true"/>
  <arg name="allow_no_texture_points"   default="false"/>

  <group ns="$(arg camera)">
    <include file="$(find realsense2_camera)/launch/includes/nodelet.launch.xml">
      <arg name="tf_prefix"                value="$(arg tf_prefix)"/>
      <arg name="external_manager"         value="$(arg external_manager)"/>
      <arg name="manager"                  value="$(arg manager)"/>
      <arg name="output"                   value="$(arg output)"/>
      <arg name="serial_no"                value="$(arg serial_no)"/>
      <arg name="usb_port_id"              value="$(arg usb_port_id)"/>
      <arg name="device_type"              value="$(arg device_type)"/>
      <arg name="json_file_path"           value="$(arg json_file_path)"/>

      <arg name="enable_pointcloud"        value="$(arg enable_pointcloud)"/>
      <arg name="pointcloud_texture_stream" value="$(arg pointcloud_texture_stream)"/>
      <arg name="pointcloud_texture_index"  value="$(arg pointcloud_texture_index)"/>
      <arg name="enable_sync"              value="$(arg enable_sync)"/>
      <arg name="align_depth"              value="$(arg align_depth)"/>

      <arg name="fisheye_width"            value="$(arg fisheye_width)"/>
      <arg name="fisheye_height"           value="$(arg fisheye_height)"/>
      <arg name="enable_fisheye"           value="$(arg enable_fisheye)"/>

      <arg name="depth_width"              value="$(arg depth_width)"/>
      <arg name="depth_height"             value="$(arg depth_height)"/>
      <arg name="enable_depth"             value="$(arg enable_depth)"/>

      <arg name="confidence_width"         value="$(arg confidence_width)"/>
      <arg name="confidence_height"        value="$(arg confidence_height)"/>
      <arg name="enable_confidence"        value="$(arg enable_confidence)"/>
      <arg name="confidence_fps"           value="$(arg confidence_fps)"/>

      <arg name="color_width"              value="$(arg color_width)"/>
      <arg name="color_height"             value="$(arg color_height)"/>
      <arg name="enable_color"             value="$(arg enable_color)"/>

      <arg name="infra_width"              value="$(arg infra_width)"/>
      <arg name="infra_height"             value="$(arg infra_height)"/>
      <arg name="enable_infra"             value="$(arg enable_infra)"/>
      <arg name="enable_infra1"            value="$(arg enable_infra1)"/>
      <arg name="enable_infra2"            value="$(arg enable_infra2)"/>
      <arg name="infra_rgb"                value="$(arg infra_rgb)"/>

      <arg name="fisheye_fps"              value="$(arg fisheye_fps)"/>
      <arg name="depth_fps"                value="$(arg depth_fps)"/>
      <arg name="infra_fps"                value="$(arg infra_fps)"/>
      <arg name="color_fps"                value="$(arg color_fps)"/>
      <arg name="gyro_fps"                 value="$(arg gyro_fps)"/>
      <arg name="accel_fps"                value="$(arg accel_fps)"/>
      <arg name="enable_gyro"              value="$(arg enable_gyro)"/>
      <arg name="enable_accel"             value="$(arg enable_accel)"/>

      <arg name="publish_tf"               value="$(arg publish_tf)"/>
      <arg name="tf_publish_rate"          value="$(arg tf_publish_rate)"/>

      <arg name="filters"                  value="$(arg filters)"/>
      <arg name="clip_distance"            value="$(arg clip_distance)"/>
      <arg name="linear_accel_cov"         value="$(arg linear_accel_cov)"/>
      <arg name="initial_reset"            value="$(arg initial_reset)"/>
      <arg name="unite_imu_method"         value="$(arg unite_imu_method)"/>
      <arg name="topic_odom_in"            value="$(arg topic_odom_in)"/>
      <arg name="calib_odom_file"          value="$(arg calib_odom_file)"/>
      <arg name="publish_odom_tf"          value="$(arg publish_odom_tf)"/>
      <arg name="allow_no_texture_points"  value="$(arg allow_no_texture_points)"/>
    </include>
  </group>
</launch>

```

Command executed
```
roslaunch realsense2_camera rs_camera.launch enable_infra:=true
```

Log
```
started roslaunch server http://robocon1-desktop:46849/

SUMMARY
========

PARAMETERS
 * /camera/realsense2_camera/accel_fps: 250
 * /camera/realsense2_camera/accel_frame_id: camera_accel_frame
 * /camera/realsense2_camera/accel_optical_frame_id: camera_accel_opti...
 * /camera/realsense2_camera/align_depth: True
 * /camera/realsense2_camera/aligned_depth_to_color_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_fisheye_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra1_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/aligned_depth_to_infra2_frame_id: camera_aligned_de...
 * /camera/realsense2_camera/allow_no_texture_points: False
 * /camera/realsense2_camera/base_frame_id: camera_link
 * /camera/realsense2_camera/calib_odom_file: 
 * /camera/realsense2_camera/clip_distance: -2.0
 * /camera/realsense2_camera/color_fps: 30
 * /camera/realsense2_camera/color_frame_id: camera_color_frame
 * /camera/realsense2_camera/color_height: 720
 * /camera/realsense2_camera/color_optical_frame_id: camera_color_opti...
 * /camera/realsense2_camera/color_width: 1280
 * /camera/realsense2_camera/confidence_fps: 30
 * /camera/realsense2_camera/confidence_height: 480
 * /camera/realsense2_camera/confidence_width: 640
 * /camera/realsense2_camera/depth_fps: 30
 * /camera/realsense2_camera/depth_frame_id: camera_depth_frame
 * /camera/realsense2_camera/depth_height: 480
 * /camera/realsense2_camera/depth_optical_frame_id: camera_depth_opti...
 * /camera/realsense2_camera/depth_width: 640
 * /camera/realsense2_camera/device_type: 
 * /camera/realsense2_camera/enable_accel: True
 * /camera/realsense2_camera/enable_color: True
 * /camera/realsense2_camera/enable_confidence: True
 * /camera/realsense2_camera/enable_depth: True
 * /camera/realsense2_camera/enable_fisheye1: False
 * /camera/realsense2_camera/enable_fisheye2: False
 * /camera/realsense2_camera/enable_fisheye: False
 * /camera/realsense2_camera/enable_gyro: True
 * /camera/realsense2_camera/enable_infra1: False
 * /camera/realsense2_camera/enable_infra2: False
 * /camera/realsense2_camera/enable_infra: True
 * /camera/realsense2_camera/enable_pointcloud: False
 * /camera/realsense2_camera/enable_pose: False
 * /camera/realsense2_camera/enable_sync: True
 * /camera/realsense2_camera/filters: 
 * /camera/realsense2_camera/fisheye1_frame_id: camera_fisheye1_f...
 * /camera/realsense2_camera/fisheye1_optical_frame_id: camera_fisheye1_o...
 * /camera/realsense2_camera/fisheye2_frame_id: camera_fisheye2_f...
 * /camera/realsense2_camera/fisheye2_optical_frame_id: camera_fisheye2_o...
 * /camera/realsense2_camera/fisheye_fps: 30
 * /camera/realsense2_camera/fisheye_frame_id: camera_fisheye_frame
 * /camera/realsense2_camera/fisheye_height: 480
 * /camera/realsense2_camera/fisheye_optical_frame_id: camera_fisheye_op...
 * /camera/realsense2_camera/fisheye_width: 640
 * /camera/realsense2_camera/gyro_fps: 400
 * /camera/realsense2_camera/gyro_frame_id: camera_gyro_frame
 * /camera/realsense2_camera/gyro_optical_frame_id: camera_gyro_optic...
 * /camera/realsense2_camera/imu_optical_frame_id: camera_imu_optica...
 * /camera/realsense2_camera/infra1_frame_id: camera_infra1_frame
 * /camera/realsense2_camera/infra1_optical_frame_id: camera_infra1_opt...
 * /camera/realsense2_camera/infra2_frame_id: camera_infra2_frame
 * /camera/realsense2_camera/infra2_optical_frame_id: camera_infra2_opt...
 * /camera/realsense2_camera/infra_fps: 30
 * /camera/realsense2_camera/infra_height: 480
 * /camera/realsense2_camera/infra_rgb: False
 * /camera/realsense2_camera/infra_width: 640
 * /camera/realsense2_camera/initial_reset: False
 * /camera/realsense2_camera/json_file_path: 
 * /camera/realsense2_camera/linear_accel_cov: 0.01
 * /camera/realsense2_camera/odom_frame_id: camera_odom_frame
 * /camera/realsense2_camera/pointcloud_texture_index: 0
 * /camera/realsense2_camera/pointcloud_texture_stream: RS2_STREAM_COLOR
 * /camera/realsense2_camera/pose_frame_id: camera_pose_frame
 * /camera/realsense2_camera/pose_optical_frame_id: camera_pose_optic...
 * /camera/realsense2_camera/publish_odom_tf: True
 * /camera/realsense2_camera/publish_tf: True
 * /camera/realsense2_camera/rosbag_filename: 
 * /camera/realsense2_camera/serial_no: 
 * /camera/realsense2_camera/tf_publish_rate: 0.0
 * /camera/realsense2_camera/topic_odom_in: odom_in
 * /camera/realsense2_camera/unite_imu_method: copy
 * /camera/realsense2_camera/usb_port_id: 
 * /rosdistro: melodic
 * /rosversion: 1.14.10

NODES
  /camera/
    realsense2_camera (nodelet/nodelet)
    realsense2_camera_manager (nodelet/nodelet)

auto-starting new master
process[master]: started with pid [10877]
ROS_MASTER_URI=http://localhost:11311

setting /run_id to e369ad62-b3fc-11eb-b8ac-204ef6c60309
process[rosout-1]: started with pid [10888]
started core service [/rosout]
process[camera/realsense2_camera_manager-2]: started with pid [10895]
process[camera/realsense2_camera-3]: started with pid [10896]
[ INFO] [1620918432.638178712]: Initializing nodelet with 6 worker threads.
[ INFO] [1620918432.938699074]: RealSense ROS v2.2.21
[ INFO] [1620918432.938840289]: Built with LibRealSense v2.41.0
[ INFO] [1620918432.938925984]: Running with LibRealSense v2.41.0
[ INFO] [1620918432.973224894]:  
[ INFO] [1620918433.047778784]: Device with serial number f1061955 was found.

[ INFO] [1620918433.047978942]: Device with physical ID /sys/devices/3610000.xhci/usb2/2-3/2-3.3/2-3.3:1.0/video4linux/video0 was found.
[ INFO] [1620918433.048093661]: Device with name Intel RealSense L515 was found.
[ INFO] [1620918433.049759402]: Device with port number 2-3.3 was found.
[ INFO] [1620918433.049918953]: Device USB type: 3.2
[ INFO] [1620918433.056089892]: getParameters...
[ INFO] [1620918433.219930803]: setupDevice...
[ INFO] [1620918433.220094034]: JSON file is not provided
[ INFO] [1620918433.220221104]: ROS Node Namespace: camera
[ INFO] [1620918433.220298383]: Device Name: Intel RealSense L515
[ INFO] [1620918433.220367343]: Device Serial No: f1061955
[ INFO] [1620918433.220437902]: Device physical port: /sys/devices/3610000.xhci/usb2/2-3/2-3.3/2-3.3:1.0/video4linux/video0
[ INFO] [1620918433.220516813]: Device FW version: 01.05.05.00
[ INFO] [1620918433.220582668]: Device Product ID: 0x0B64
[ INFO] [1620918433.220681579]: Enable PointCloud: Off
[ INFO] [1620918433.220843273]: Align Depth: On
[ INFO] [1620918433.220946472]: Sync Mode: On
[ INFO] [1620918433.221102502]: Device Sensors: 
[ INFO] [1620918433.224168933]: L500 Depth Sensor was found.
[ INFO] [1620918433.231977838]: RGB Camera was found.
[ INFO] [1620918433.232762566]: Motion Module was found.
[ INFO] [1620918433.232932836]: num_filters: 0
[ INFO] [1620918433.233043331]: Setting Dynamic reconfig parameters.
[ INFO] [1620918433.362056690]: Done Setting Dynamic reconfig parameters.
[ INFO] [1620918433.364120411]: depth stream is enabled - width: 640, height: 480, fps: 30, Format: Z16
[ INFO] [1620918433.364726901]: infra stream is enabled - width: 640, height: 480, fps: 30, Format: Y8
[ INFO] [1620918433.368250766]: color stream is enabled - width: 1280, height: 720, fps: 30, Format: RGB8
[ INFO] [1620918433.368998533]: confidence stream is enabled - width: 640, height: 480, fps: 30, Format: RAW8
[ WARN] [1620918433.374678247]: No mathcing profile found for accel with fps=250
[ WARN] [1620918433.374842981]: profiles found for accel:
[ WARN] [1620918433.374924708]: fps: 400. format: MOTION_XYZ32F
[ WARN] [1620918433.375048547]: fps: 200. format: MOTION_XYZ32F
[ WARN] [1620918433.375142626]: fps: 100. format: MOTION_XYZ32F
[ INFO] [1620918433.375295168]: setupPublishers...
[ INFO] [1620918433.384883510]: Expected frequency for depth = 30.00000
[ INFO] [1620918433.476718592]: Expected frequency for infra = 30.00000
[ INFO] [1620918433.537282467]: Expected frequency for aligned_depth_to_infra = 30.00000
[ INFO] [1620918433.599546548]: Expected frequency for color = 30.00000
[ INFO] [1620918433.660883278]: Expected frequency for aligned_depth_to_color = 30.00000
[ INFO] [1620918433.721179957]: Expected frequency for confidence = 30.00000
[ INFO] [1620918433.780981504]: setupStreams...
[ INFO] [1620918433.835060875]: insert Depth to L500 Depth Sensor
[ INFO] [1620918433.835536806]: insert Color to RGB Camera
[ INFO] [1620918433.835755907]: insert Infrared to L500 Depth Sensor
[ INFO] [1620918433.835974753]: insert Gyro to Motion Module
[ INFO] [1620918433.836117823]: insert Confidence to L500 Depth Sensor
 13/05 23:07:14,114 WARNING [547821773184] (ds5-motion.cpp:477) IMU Calibration is not available, default intrinsic and extrinsic will be used.
[ INFO] [1620918434.130634662]: SELECTED BASE:Depth, 0
[ INFO] [1620918434.187086562]: RealSense Node Is Up!
 13/05 23:07:14,426 WARNING [547376279936] (l500-depth.cpp:455) UVC metadata payloads are not available for stream . Please refer to installation chapter for details.
 13/05 23:07:15,048 WARNING [547643969920] (l500-depth.cpp:455) UVC metadata payloads are not available for stream . Please refer to installation chapter for details.
[ERROR] [1620918435.104512032]: An error has occurred during frame callback: map::at
[ERROR] [1620918435.144468983]: An error has occurred during frame callback: map::at
[ERROR] [1620918435.163327119]: An error has occurred during frame callback: map::at
[ERROR] [1620918435.199078354]: An error has occurred during frame callback: map::at
[ERROR] [1620918435.244310161]: An error has occurred during frame callback: map::at
```
# Approach tried (but failed)

From #1348, using `roslaunch realsense2_camera rs_camera.launch enable_infra:=true` did not solve the issue. The error is still here.

Rebooting the NX without replugging the L515 will make the error message gone, but then the images in the `align_depth` topics are not showing anymore.

**Comment by doronhi** [comment]
This error message seems like a bug. Before I start looking into it, could you confirm it happens with the latest version as well? (realsense2_camera version 2.3.0)

**Comment by vinesmsuic** [comment]
Removed all the realsense packages, installed the lastest RealSenseSDK v2.45.0 (build from source) and realsense-ros v2.3.0, now the error is gone and everything works perfectly.

Another issue I found is that the ddynamic-reconfigure  package must be cloned to src folder to get the realsense-ros successfully build. Using APT to install dynamic-reconfigure and ddynamic-reconfigure will not work at all.
